File: backend/src/main/kotlin/org/globe42/dao/CountryDao.java
package org.globe42.dao;

import java.util.List;

import org.globe42.domain.Country;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

/**
 * DAO for the {@link Country} entity
 * @author JB Nizet
 */
public interface CountryDao extends JpaRepository<Country, String> {
    @Query("select c from Country c order by c.name")
    List<Country> findAllSortedByName();
}

File: backend/src/main/kotlin/org/globe42/dao/CountryDao.kt
package org.globe42.dao

import org.globe42.domain.Country
import org.springframework.data.jpa.repository.Query

/**
 * DAO for the [Country] entity
 * @author JB Nizet
 */
interface CountryDao : GlobeRepository<Country, String> {
    @Query("select c from Country c order by c.name")
    fun findAllSortedByName(): List<Country>
}
